cpp/src/arrow/util/functional.h

```diff
@@ -73,6 +73,10 @@ struct call_traits {
 
   template <typename F>
   using return_type = decltype(return_type_impl(&std::decay<F>::type::operator()));
+
+  template <typename F, typename T, typename RT = T>
+  using enable_if_return_type =
```

Comment:
```suggestion
  using enable_if_return_type_is =
```
```suggestion
  using enable_if_return =
```
Purely because the `_type` suffix has the specific connotation of "a metafunction which returns some type"

Reply:
Good point, will do.